### PR TITLE
- Use a single store for mapping lid to its data that are split into …

### PR DESCRIPTION
**Pull request by baldersheim**
…partitions and chunks.

- This enable memory to be released after compaction is done.

@geirst PR

### CODE REVIEW

File: searchlib/src/tests/docstore/store_by_bucket/store_by_bucket_test.cpp [hunk 1]
```diff
@@ -56,33 +57,61 @@ class VerifyBucketOrder : public StoreByBucket::IWrite {
...
         add(sbb, i);
     }
     for (size_t i(1000); i > 500; i--) {
         add(sbb, i);
     }
     sbb.close();
-    EXPECT_EQUAL(32u, sbb.getBucketCount());
-    EXPECT_EQUAL(1000u, sbb.getLidCount());
+    std::sort(storeIndex._where.begin(), storeIndex._where.end());
+    //EXPECT_EQUAL(32u, sbb.getBucketCount());
```
Comment (geirst) [inline searchlib/src/tests/docstore/store_by_bucket/store_by_bucket_test.cpp:110]: Why is this commented out?

File: searchlib/src/vespa/searchlib/docstore/storebybucket.h [hunk 2]
```diff
@@ -23,51 +23,55 @@ class StoreByBucket
...
+    struct Index {
+        using BucketId=document::BucketId;
+        Index(BucketId bucketId, uint32_t id, uint32_t chunkId, uint32_t entry) noexcept :
+                _bucketId(bucketId), _id(id), _chunkId(chunkId), _lid(entry)
+        { }
+        bool operator < (const Index & b) const noexcept {
+            return BucketId::bucketIdToKey(_bucketId.getRawId()) < BucketId::bucketIdToKey(b._bucketId.getRawId());
+        }
+        BucketId _bucketId;
+        uint32_t _id;
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/docstore/storebybucket.h:35]: Consider a more descriptive name.

**Review by geirst — COMMENTED** [review]
👍  But we need direct unit testing of BucketCompacter before merging this.

File: searchlib/src/tests/docstore/store_by_bucket/store_by_bucket_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/tests/docstore/store_by_bucket/store_by_bucket_test.cpp]: It does not exist on this level anymore.
See new test of BucketIndexStore.

File: searchlib/src/vespa/searchlib/docstore/storebybucket.h (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/docstore/storebybucket.h]: id -> localChunkId

**Comment by baldersheim** [comment]
@geirst PTAL

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**